src/solvers/smt2_incremental/smt_solver_process.h

```diff
@@ -35,9 +37,12 @@ class smt_piped_solver_processt : public smt_base_solver_processt
   ///   The command and arguments for invoking the smt2 solver.
   /// \param message_handler:
   ///   The messaging system to be used for logging purposes.
+  /// \param out_stream:
+  /// Pointer to the stream to print the SMT formula. `nullptr` if no output.
   smt_piped_solver_processt(
     std::string command_line,
-    message_handlert &message_handler);
+    message_handlert &message_handler,
+    std::unique_ptr<std::ostream> out_stream);
```

Comment:
Is passing the `std::unique_ptr` by value rather than as a reference the right thing to do?

Reply:
Yes as we want to extend the lifetime of the passed stream to the one of the `smt_solver_proces` object that is longer than the one of the `solver_factory` where the pointer is created.